scripts/o3de/o3de/github_utils.py

```diff
@@ -61,9 +76,18 @@ def clone_from_git(uri, download_path: pathlib.Path) -> int:
             logger.error(str(e))
             return 1
 
+        if proc.returncode == 0 and ref:
+            params = ["git", "-C", download_path.as_posix(), "reset", "--hard", ref]
+            try:
+                with subprocess.Popen(params, stdout=subprocess.PIPE) as proc:
+                    print(proc.stdout.read())
+            except Exception as e:
+                logger.error(str(e))
+                return 1
+
         return proc.returncode
 
-def get_github_provider(parsed_uri) -> GitHubProvider or None:
+def get_github_provider(parsed_uri:ParseResult) -> GitHubProvider or None:
```

Comment:
nitpick: missing whitespace after color
```suggestion
def get_github_provider(parsed_uri: ParseResult) -> GitHubProvider or None:
```